src/cljs/pyregence/pages/near_term_forecast.cljs

```
@@ -824,46 +824,39 @@
-                                                      .getBoundingClientRect
-                                                      (aget "height")))
-                                               "px"))
-                          (js/setTimeout mb/resize-map! 50))]
-          (-> js/window (.addEventListener "touchend" update-fn))
-          (-> js/window (.addEventListener "resize"   update-fn))
-          (initialize! params)
-          (update-fn)))
-      :reagent-render
-      (fn [_]
-        [:div#near-term-forecast
-         {:style ($/combine $/root {:height @height :padding 0 :position "relative" :overflow :hidden})}
-         [message-box-modal]
-         (when @!/loading? [loading-modal])
-         [message-modal]
-         [nav-bar {:capabilities     @!/capabilities
-                   :current-forecast @!/*forecast
-                   :is-admin?        (->> @!/user-orgs-list
-                                          (filter #(= "admin" (:role %)))
-                                          (count)
-                                          (< 0)) ; admin of at least one org
-                   :logged-in?       user-id
-                   :mobile?          @!/mobile?
-                   :user-orgs-list   @!/user-orgs-list
-                   :select-forecast! select-forecast!
-                   :user-id          user-id}]
-         [:div {:style {:height "100%" :position "relative" :width "100%"}}
-          (when (and @mb/the-map
-                     (not-empty @!/capabilities)
-                     (not-empty @!/*params))
-            [control-layer user-id])
-          [map-layer]
-          [pop-up]]])})))
+  (r/create-class
+   {:component-did-mount
+    (fn [_]
+      (let [update-fn (fn [& _]
+                        (-> js/window (.scrollTo 0 0))
+                        (reset! !/mobile? (> 800.0 (.-innerWidth js/window)))
+                        (js/setTimeout mb/resize-map! 50))]
```

Do we still need this? Jusk asking, since the map resizes on load now.

---

@danielhvs the short answer is: i don't know because i don't understand why the timeout was there in the first place. This is kind of the problem with timeouts, its very unclear what they are trying to communicate. 

I think that resize-map! function can go away regardless, i feel like i would rather it just throw an error if the map isn't there and we can't call re-size. I don't understand the idea of returning nil and ignoring the error. Unless it somehow halts the program or something?

---

Just to make a comment, I'm not asking for changes but feel free to do so:
Agreed. I think we can drop the (when ...) and just call .resize and if it fails, something is wrong and we can see it, rather than throwing dust below the carpet.